### PR TITLE
Backport CMake changes

### PR DESCRIPTION
**Pull request by dnakamura**
ibmruntimes/openj9-openjdk-jdk#153
ibmruntimes/openj9-openjdk-jdk#186
ibmruntimes/openj9-openjdk-jdk#190
ibmruntimes/openj9-openjdk-jdk#192

Signed-off-by: Devin Nakamura <devinn@ca.ibm.com>

### CODE REVIEW

File: jdk/make/closed/autoconf/custom-hook.m4 [hunk 1]
```diff
@@ -306,35 +306,31 @@ AC_DEFUN([OPENJ9_CHECK_NASM_VERSION],
...
+      # ---
+      # NASM version 2.14.02 compiled on Dec 27 2018
+      # ---
+      # Some builds may not contain any text after the version number
+      #
+      # NASM_VERSION is set within square brackets so that the sed expression would not
+      # require quadrigraps to represent square brackets
+      [NASM_VERSION=`$NASM -v | $SED -e 's/^.* \([2-9]\.[0-9][0-9]\.[0-9][0-9]\).*$/\1/'`]
+      AC_MSG_ERROR([nasm version detected: $NASM_VERSION; required version 2.11+])
     fi
```
Comment (keithc-ca) [inline jdk/make/closed/autoconf/custom-hook.m4:334]: This is missing:
```
AC_SUBST([NASM])
```

File: closed/OpenJ9.gmk [hunk 2]
```diff
@@ -247,20 +247,28 @@ CMAKE_ARGS := \
...
-endif
+  ifneq (,$(OPENJ9_CXX))
+    CMAKE_ARGS += -DCMAKE_CXX_COMPILER="$(OPENJ9_CXX)"
+  else
+    CMAKE_ARGS += -DCMAKE_CXX_COMPILER="$(ac_cv_prog_CXX)"
+  endif
+
+  ifneq (,$(NASM))
+    CMAKE_ARGS += -DCMAKE_ASM_NASM_COMPILER="$(NASM)"
+  endif
```
[the remark this answers is not in the document]
Comment (dnakamura) [reply closed/OpenJ9.gmk:270]: Didn't think of it, updating the other PRs

File: jdk/make/closed/autoconf/toolchain-win.cmake.in [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+if("@OPENJDK_TARGET_CPU_BITS@" EQUAL "64")
+	set(CMAKE_ASM_MASM_COMPILER "${tool_dir}/ml64" CACHE FILEPATH "")
+else()
+	set(CMAKE_ASM_MASM_COMPILER "${tool_dir}/ml" CACHE FILEPATH "")
+endif()
+set(CMAKE_ASM_NASM_COMPILER "${tool_dir}/nasm" CACHE FILEPATH "")
+
+set(CMAKE_Java_AR "${tool_dir}/jar" CACHE FILEPATH "")
+set(CMAKE_Java_RUNTIME "${tool_dir}/java" CACHE FILEPATH "")
+set(CMAKE_Java_COMPILER "${tool_dir}/javac" CACHE FILEPATH "")
```
[the remark this answers is not in the document]
Comment (dnakamura) [reply jdk/make/closed/autoconf/toolchain-win.cmake.in:42]: See ibmruntimes/openj9-openjdk-jdk#192

File: jdk/make/closed/autoconf/toolchain-win.cmake.in [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+separate_arguments(FIXPATH UNIX_COMMAND "@FIXPATH@")
+set(OMR_EXE_LAUNCHER "${FIXPATH}" CACHE STRING "")
```
[the remark this answers is not in the document]
Comment (dnakamura) [reply jdk/make/closed/autoconf/toolchain-win.cmake.in:45]: same as above

**Comment by keithc-ca** [comment]
Jenkins test sanity plinux,xlinuxcm,win,win32 jdk8

**Pull request merged**